File: src/AlphabetSpam/AlphabetSpam.java
import java.util.Scanner;

public class AlphabetSpam {
    public static void main(String[] args) {
        Double[] count = new Double[] {0d, 0d, 0d, 0d};
        String text;

        try (Scanner input = new Scanner(System.in)) {
            text = input.nextLine();
        }

        for (char i : text.toCharArray()) {
            int num = (int) i;
            if (num == 95) {
                count[0]++;
            } else if ((num >= 97) && (num <= 122)) {
                count[1]++;
            } else if ((num >= 65) && (num <= 90)) {
                count[2]++;
            } else {
                count[3]++;
            }
        }

        for (double i : count) {
            System.out.println(i / text.length());
        }
    }
}

File: src/AlphabetSpam/alphabetspam.py
# variable setup: set all counters to zero
space, lower, upper, symbol = 0, 0, 0, 0
text = input()

for i in text:
    num = ord(i)
    # using ascii values to split apart four types
    if num == 95:
        space += 1
    elif 97 <= num <= 122:
        lower += 1
    elif 65 <= num <= 90:
        upper += 1
    else:
        # known: input will be between ascii 33-126
        symbol += 1

# iterate through four values and print out ratios
for count in (space, lower, upper, symbol):
    print(count / len(text))